### ISSUE TITLE
Enhance AggregationFuzzer to verify results against Spark

### ISSUE THREAD

**Issue by rui-mo** [facebookincubator/velox#9270]
### Description

Currently, Aggregation Fuzzer verifies results against DuckDB. However, not all functions are available in DuckDB and sometimes semantics don't match. It would be better to verify against Spark.

**Comment by rui-mo** [comment]
https://github.com/facebookincubator/velox/issues/6595#issuecomment-1829141664

**Comment by rui-mo** [comment]
Let's track the updates of SparkQueryRunner here. cc: @mbasmanova @zhztheplayer

**Comment by mbasmanova** [comment]
CC: @kgpai @kagamiori

**Comment by mbasmanova** [comment]
CC: @amitkdutta

**Comment by rui-mo** [comment]
Hi @mbasmanova, I have updated our recent updates in this issue. Your feedback is appreciated, thanks. cc: @FelixYBW

**Comment by mbasmanova** [comment]
@rui-mo Rui, this is great. Looks like you have a working prototype. What would be the next steps towards "productizing" this?

CC: @kgpai @assignUser @duanmeng

**Comment by rui-mo** [comment]
@mbasmanova If this approach makes sense in Velox, we plan to introduce the Spark client as well as the Spark connect protocols into Velox, and start to enhance the aggregation fuzzer based on them. To test against Spark, we would like to know where we can set-up the Spark environment needed for execution.

**Comment by assignUser** [comment]
> where we can set-up the Spark environment needed for execution
What's the requirement for this? I assume this would be more inline of an ephemeral test setup  (e.g. how we use hadoop on the adapters test) and not a permanent production grade thing?

**Comment by rui-mo** [comment]
> I assume this would be more inline of an ephemeral test setup

@assignUser Yes, you are right. Could you provide us a reference on how hadoop or presto is deployed in the test? I assume we can deploy spark in a similar way. Thank you.

**Comment by assignUser** [comment]
It looks like the tests themselves start hadoop/azurite for example see `connectors/hive/storage_adapters/hive/tests`
If you show me how you setup spark for your poc tests (e.g. your bash script) I can help with getting that setup in an action. 

Here is the service container reference, sounds useful https://docs.github.com/en/actions/using-containerized-services/about-service-containers
